roxie/topo/toposerver.cpp

```diff
@@ -300,16 +302,51 @@ int main(int argc, const char *argv[])
         else
             globals->loadProp(argv[i], true);  // We ignore unrecognized options for now
     }
+
+    // locate settings xml file in runtime dir
+    char currentDirectory[_MAX_DIR];
+    if (!getcwd(currentDirectory, sizeof(currentDirectory)))
+    {
+        perror("getcwd failure");
+        return EXIT_FAILURE;
+    }
+    topologyFile.set(currentDirectory);
+    addNonEmptyPathSepChar(topologyFile);
+    topologyFile.append(PATHSEPCHAR).append("toposerver.xml");
+    IPropertyTree *topology;
+    if (checkFileExists(topologyFile.str()))
+    {
+        topology = createPTreeFromXMLFile(topologyFile.str(), ipt_lowmem);
+        if (globals->hasProp("--traceLevel"))
+            topology->setProp("@traceLevel", globals->queryProp("--traceLevel"));
+        if (globals->hasProp("--port"))
+            topology->setProp("@port", globals->queryProp("--port"));
+        if (globals->hasProp("--stdlog"))
+            topology->setProp("@stdlog", globals->queryProp("--stdlog"));
+        if (globals->hasProp("--logdir"))
+            topology->setProp("@logdir", globals->queryProp("--logdir"));
+    }
+    else
+    {
+        topology = createPTreeFromXMLString(
+            "<TopoServerProcess port='9004' traceLevel='1'>"
+            "</TopoServerProcess>"
+            , ipt_lowmem
+            );
+    }
+
+
     // take ownership of sentinel file
     Owned<IFile> sentinelFile = createSentinelTarget();
     removeSentinelFile(sentinelFile);
-    traceLevel = globals->getPropInt("--traceLevel", 1);
```

Comment:
You have removed the ability to override settings from the command line... These should override anything read from the topology file.

Reply:
moved the section that overwrites the topology with globals.  Overrides correctly now.